src/py/flwr/server/workflow/default_workflows.py

```diff
@@ -115,40 +116,47 @@ def default_init_params_workflow(driver: Driver, context: Context) -> None:
             parameters, keep_input=True
         )
     else:
-        # Get initial parameters from one of the clients
-        log(INFO, "Requesting initial parameters from one random client")
-        random_client = context.client_manager.sample(1)[0]
-        # Send GetParametersIns and get the response
-        content = compat.getparametersins_to_recordset(GetParametersIns({}))
-        messages = driver.send_and_receive(
-            [
-                driver.create_message(
-                    content=content,
-                    message_type=MessageTypeLegacy.GET_PARAMETERS,
-                    dst_node_id=random_client.node_id,
-                    group_id="0",
-                )
-            ]
-        )
-        log(INFO, "Received initial parameters from one random client")
-        msg = list(messages)[0]
-        paramsrecord = next(iter(msg.content.parameters_records.values()))
+        while True:
+            # Get initial parameters from one of the clients
+            log(INFO, "Requesting initial parameters from one random client")
+            random_client = context.client_manager.sample(1)[0]
+            # Send GetParametersIns and get the response
+            content = compat.getparametersins_to_recordset(GetParametersIns({}))
+            messages = driver.send_and_receive(
+                [
+                    driver.create_message(
+                        content=content,
+                        message_type=MessageTypeLegacy.GET_PARAMETERS,
+                        dst_node_id=random_client.node_id,
+                        group_id="0",
+                    )
+                ]
+            )
+            log(INFO, "Received initial parameters from one random client")
```

Comment:
Should we move this inside the `if msg.has_content()`?

Reply:
Good point